server/src/DBTablesMonitoring.cc

```diff
@@ -1198,6 +1198,15 @@ string TriggersQueryOption::getCondition(void) const
 			MONITORING_SERVER_SELF_ID);
 	}
 
+	if (m_impl->triggerCollectHost >= REAL_HOST_TRIGGER) {
```

Comment:
I feel it's hard to get what we'd like to do.
How about add a comment like this.
// We only correct valid triggers (i.e. existing in the Zabbix server)
// including self monitoring triggers.

or should we use a readable method name or variable names like
if (correctOnlyValidTriggersIncludingSelfMonitors()) {
}